### DISCUSSION TITLE
Problem with usage

### DISCUSSION THREAD

**Discussion by ponasromas** [spatie/dns#82] [General]
Having this as a test:

```
<?php

use Spatie\Dns\Dns;

$dns = new Dns();

$dns->getRecords('google.com', 'MX');
```
Getting error:

> Fatal error: Uncaught ArgumentCountError: Too few arguments to function Spatie\Dns\Dns::__construct(), 0 passed in<...> at least 1 expected in /vendor/spatie/dns/src/Dns.php:34 Stack trace: #0 <...> Spatie\Dns\Dns->__construct() #1 {main} thrown in /vendor/spatie/dns/src/Dns.php on line 34

Why new Dns() requires domain when examples in documentation does not mention it? Even with domain it spits error about invalid domain.

**Comment by freekmurze** [comment]
As you can see here, there are no required arguments for the constructor: https://github.com/spatie/dns/blob/7aa3d3f465966173bc6fb15f86ceaf0528dd720c/src/Dns.php#L27-L33

Probably you're not using the latest version.

**Comment by ponasromas** [comment]
Ha, you are absolutely correct. spatie package was referred by another DNS package :O . My mistake, sorry.

**Reply by joho1968** [reply]
Though, with a "fresh" composer install, you run into the exact same problem, or at least I did.

`composer require spatie/dns`

Installs version "1.6.0 - 2020-12-02" (according to changelog.md).

Possibly because I'm using PHP 7.4.x, in which case it's fine/correct :-) But it may be worth mentioning somewhere other than in CHANGELOG.md.
